tools/gfx/render-d3d12.cpp

```diff
@@ -1388,32 +1339,95 @@ Result D3D12Renderer::initialize(const Desc& desc, void* inWindowHandle)
         }
     }
 
-    if (!adapter)
+    if (m_dxDebug && (deviceCheckFlags & DeviceCheckFlag::UseDebug))
     {
-        // Couldn't find an adapter
+        m_dxDebug->EnableDebugLayer();
+    }
+
+    outDxgiFactory = dxgiFactory;
+    outAdapter = adapter;
+    return SLANG_OK;
+}
+
+Result D3D12Renderer::initialize(const Desc& desc, void* inWindowHandle)
+{
+    m_hwnd = (HWND)inWindowHandle;
+    // Rather than statically link against D3D, we load it dynamically.
+
+    HMODULE d3dModule = LoadLibraryA("d3d12.dll");
+    if (!d3dModule)
+    {
+        fprintf(stderr, "error: failed load 'd3d12.dll'\n");
         return SLANG_FAIL;
     }
 
-    // set up debug layer
-#ifndef NDEBUG
+    HMODULE dxgiModule = LoadLibraryA("Dxgi.dll");
+    if (!dxgiModule)
     {
+        fprintf(stderr, "error: failed load 'dxgi.dll'\n");
```

Comment:
Should we be piping the I/O redirection stuff into here?

Reply:
We need to come up with a mechanism for reporting errors for devices startup. The thing that caused the PR to be made was, on certain CI machines tests were failing in renderer startup and was not clear why.